### PR TITLE
Feature/dev env

### PR DESCRIPTION
**Pull request by modkaffes**
This simplifies the setup procedure a bit, making it easier for new maintainers to run the webapp in their local envs.

### CODE REVIEW

**Comment by guylepage3** [comment]
Reviewing this now..

File: README.md [hunk 1]
```diff
@@ -79,24 +79,11 @@ Install [`npm`](https://www.npmjs.com/get-npm). `npm` is installed and comes bun
...
-```
-$ cd client
-$ npm install
-```
-
-Go back to the server
-
-```
-$ cd ../
+$ npm install && npm run client-install
```
Comment (guylepage3) [inline README.md]: Hey @modkaffes, let's change these changes back as it is easier for beginner devs to understand what's going on. Thx

**Comment by guylepage3** [comment]
@modkaffes is this relevant as well? I went to port 4000 after running the app and I'm assuming that it was running some sort of server as I didn't get an error. But we should have it run on 4000 or 5000..

Screenshot 2019-06-07 14 26 33

File: README.md (hunk 1 above)
Comment (modkaffes) [reply README.md]: Ah, I saw the `client-install` script in place and thought we could use this for brevity. Should I remove `client-install` as well and just use the old sequence? (cd in, install, cd back)

**Comment by modkaffes** [comment]
@guylepage3 this is [json-server](https://github.com/typicode/json-server) at play.

I wired it up to provide a mock api server at port 4000 by [this script](https://github.com/universelabs/universe-webapp/pull/3/files#diff-52dbbf94d5c54b68c0cad0702ba94a96R29). It gives you a mock server you can use while developing (and hit it with curl, postman etc).

Port is definitely interchangable.

File: README.md (hunk 1 above)
Comment (guylepage3) [reply README.md]: Yeah I thought about having everything run for the developer but I think we should have the developer run the server and then the client manually.

**Comment by guylepage3** [comment]
I see.. so we have two servers running. That’s something that we should simplify for the developer so that they can run one command and it runs both servers.. but we should let the developer know that this taking place both in a comment above the script in `package.json` as well in the terminal as both servers run.

**Comment by modkaffes** [comment]
There’s three things needed for the app to work in its current state:

1. The express server (which serves the routes, `/`, `/subscribe`, etc)
2. The `client` app (which serves the ui)
3. A data server which sends the keys data to be displayed by the ui. Normally this should be our backend api, but as it isn’t ready yet I wired up this `json-server` in order the ui development is independent until api is ready.

This is all being handled by [this script](https://github.com/universelabs/universe-webapp/pull/4/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R33) which boots up express app, mock data and client app all in one. Should I separate the scripts?

**Comment by guylepage3** [comment]
Yes. Now that I think about it..

We should always run the server(s) manually away from the client. We can build a script in `package.json` to run all the servers but we should let the developer know that they are running and which port they are running on.. ie...

Developer runs
```
$ npm run servers
```

Output in Terminal

```
Temporary Data Server listening on 4000
Starting the temporary data server...

Loading temporary data source... ./src/api/db.json
Loading done...

  Resources:            http://localhost:4000/data
  Home:                 http://localhost:4000/

Development Server listening on 5000
Starting the development server...
```

Then the developer would run the client in another tab in Terminal.

The reason we want them to manually run the servers and then run the client manually is so that they can see which servers are running and what they do and then they can debug from there. After the servers are running smoothly, then they can run the front end client.

File: README.md (hunk 1 above)
Comment (modkaffes) [reply README.md]: Definitely, rolling it back after offline discussion.

**Comment by guylepage3** [comment]
Reviewing now

**Comment by guylepage3** [comment]
@modkaffes FYI, I made a small update to your script. Always want to avoid using shift key when possible... Changed `dev:server` to `dev-server`.

**Comment by modkaffes** [comment]
> @modkaffes FYI, I made a small update to your script. Always want to avoid using shift key when possible... Changed `dev:server` to `dev-server`.

Of course, not a problem at all. The colon namespacing was a habit from my gulp days 😀

**Pull request merged**